zipline/finance/commission.py

```diff
@@ -69,6 +69,18 @@ def calculate(self, order, transaction):
         """
         raise NotImplementedError('calculate')
 
+    def __eq__(self, other):
+        return self.asdict() == other.asdict()
```

Comment:
The conclusion from all of the above, I think, is that there's no safe and general way to implement `__eq__` in terms of the instance members of an arbitrary subclass.  If we want to be able to equality-compare any two instances of CommissionModel, then that imposes constraints on what a subclass of CommissionModel can do, which means we probably need tests for each subclass of CommissionModel to ensure that they actually abide by those constraints.  In particular, as noted above, I think `__eq__` is broken right now for `PerContract`.